matsim/src/test/java/org/matsim/pt/utils/PtNetworkSimplifierTest.java

```diff
@@ -0,0 +1,184 @@
+package org.matsim.pt.utils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.matsim.api.core.v01.Coord;
+import org.matsim.api.core.v01.Id;
+import org.matsim.api.core.v01.Scenario;
+import org.matsim.api.core.v01.TransportMode;
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.api.core.v01.network.Node;
+import org.matsim.core.config.ConfigUtils;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.core.network.algorithms.NetworkCalcTopoType;
+import org.matsim.core.population.routes.NetworkRoute;
+import org.matsim.core.population.routes.RouteUtils;
+import org.matsim.core.scenario.ScenarioUtils;
+import org.matsim.core.utils.geometry.CoordUtils;
+import org.matsim.pt.transitSchedule.api.TransitLine;
+import org.matsim.pt.transitSchedule.api.TransitRoute;
+import org.matsim.pt.transitSchedule.api.TransitRouteStop;
+import org.matsim.pt.transitSchedule.api.TransitSchedule;
+import org.matsim.pt.transitSchedule.api.TransitScheduleFactory;
+import org.matsim.pt.transitSchedule.api.TransitStopFacility;
+
+public class PtNetworkSimplifierTest {
+
+	@Test
+	public void testPtNetworkSimplifier() {
+		Scenario scenario = buildScenario();
+
+		Set<Integer> nodeTypesToMerge = new TreeSet<>();
+		nodeTypesToMerge.add(NetworkCalcTopoType.PASS1WAY);
+		nodeTypesToMerge.add(NetworkCalcTopoType.PASS2WAY);
+
+		PtNetworkSimplifier ptSimplifier = new PtNetworkSimplifier(scenario.getNetwork(),
+				scenario.getTransitSchedule());
+		ptSimplifier.setNodesToMerge(nodeTypesToMerge);
+		ptSimplifier.run();
+
+		// remove unused nodes
+		Set<Id<Node>> nodeIdsToRemove = new HashSet<>();
+		for (Node node : scenario.getNetwork().getNodes().values()) {
+			if (node.getInLinks().isEmpty() && node.getOutLinks().isEmpty()) {
+				nodeIdsToRemove.add(node.getId());
+			}
+		}
+		nodeIdsToRemove.stream().forEach(scenario.getNetwork()::removeNode);
+		
+		Assert.assertFalse(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("FG")));
+		Assert.assertFalse(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("GH")));
+		Assert.assertFalse(scenario.getNetwork().getNodes().keySet().contains(Id.createNodeId("G")));
+		Assert.assertTrue(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("FG-GH")));
+		Assert.assertFalse(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("CX")));
+		Assert.assertFalse(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("XY")));
+		Assert.assertFalse(scenario.getNetwork().getNodes().keySet().contains(Id.createNodeId("X")));
+		Assert.assertTrue(scenario.getNetwork().getLinks().keySet().contains(Id.createLinkId("CX-XY")));
+	}
```

Comment:
It would be nice to test the resulting transit lines, too.

Reply:
Sure, I'll add a test for the transit lines throughout this week.

Reply:
@vsp-gleich Also checked the links in the transit route.